code/modules/mob/living/carbon/human/species/station.dm

```diff
@@ -378,11 +384,155 @@
 	has_organ = list(
 		"brain" = /obj/item/organ/brain/slime
 		)
-		
+
 	suicide_messages = list(
 		"is melting into a puddle!",
-		"is turning a dull, brown color and melting into a puddle!",
-		"is ripping out their own core!")
+		"is ripping out their own core!",
+		"is turning a dull, brown color and melting into a puddle!")
+
+	var/list/mob/living/carbon/human/recolor_list = list()
+
+	species_abilities = list(
+		/mob/living/carbon/human/verb/toggle_recolor_verb,
+		/mob/living/carbon/human/proc/regrow_limbs
+		)
+
+/datum/species/slime/handle_life(var/mob/living/carbon/human/H)
+	var/const/color_shift_trigger = 0.1
+	var/const/icon_update_period = 600 // 1 minute
+	var/const/blood_scaling_factor = 5 // Used to adjust how much of an effect the blood has on the rate of color change. Higher is slower.
+	// Slowly shifting to the color of the reagents
+	if((H in recolor_list) && H.reagents.total_volume > color_shift_trigger)
+		var/blood_amount = H.vessel.total_volume
+		var/r_color = mix_color_from_reagents(H.reagents.reagent_list)
+		var/new_body_color = BlendRGB(r_color, rgb(H.r_skin, H.g_skin, H.b_skin), (blood_amount*blood_scaling_factor)/((blood_amount*blood_scaling_factor)+(H.reagents.total_volume)))
+		var/list/new_color_list = ReadRGB(new_body_color)
+		H.r_skin = new_color_list[1]
+		H.g_skin = new_color_list[2]
+		H.b_skin = new_color_list[3]
+		if(world.time % icon_update_period > icon_update_period - 20) // The 20 is because this gets called every 2 seconds, from the mob controller
+			for(var/organname in H.organs_by_name)
+				var/obj/item/organ/external/E = H.organs_by_name[organname]
+				if(istype(E) && E.dna.species == "Slime People")
+					E.sync_colour_to_human(H)
+			H.regenerate_icons() // Because update_icon didn't work
```

Comment:
It might also help to distribute this out - should I add, perhaps, a hash of the slime person's memory location or similar, and use that to stagger out the icon updates?